### PR TITLE
Modified challenge create page

### PR DESCRIPTION
**Pull request by sanketbansal**
#### Changes proposed in this pull request:

- Changes made in UI to make it look similar to current Eval AI application.
- Changes made in challenge service to subscribe for api callback of createChallenge method.
- Changes made in the logic to handle the file upload and show related errors and toast.

- Link to live demo: http://pr-150-evalai.surge.sh 

Screenshot from 2019-06-16 10-49-36

Screenshot from 2019-06-16 10-49-54

### CODE REVIEW

**Comment by sanketbansal** [comment]
@lunayach @Shekharrajak please review the PR.

**Comment by Shekharrajak** [comment]
Similar commits -- https://github.com/Cloud-CV/EvalAI-ngx/pull/149#issuecomment-503590467

**Comment by Shekharrajak** [comment]
Still 37 file changes. I can't review it. Also, I see changes like other PRs.

**Comment by sanketbansal** [comment]
> Still 37 file changes. I can't review it. Also, I see changes like other PRs.

The file changes will get further reduce after merging #141 . I kept these changes as these are required for this PR also.

**Comment by Shekharrajak** [comment]
Okay thanks for letting me know.

**Comment by sanketbansal** [comment]
@lunayach this can be reviewed also.

**Comment by lunayach** [comment]
@sanketbansal can you confirm if you are able to create a host team on your end. When I navigate to `Create Challenge`, it redirects me to `/teams/hosts`. There, just an empty page appears.

**Comment by sanketbansal** [comment]
> @sanketbansal can you confirm if you are able to create a host team on your end. When I navigate to `Create Challenge`, it redirects me to `/teams/hosts`. There, just an empty page appears.

I think that issue should be resolved after merging #149

File: src/app/components/challenge-create/challenge-create.component.scss [hunk 1]
```diff
@@ -1,38 +1,72 @@
...
   }
-  .submit-btn-div {
-    .btn {
-      margin-left:0;
-    }
+  &:after {
+    content: '';
+    position: relative;
+    display: inline-block;
+    color: black;
```
Comment (lunayach) [inline src/app/components/challenge-create/challenge-create.component.scss]: Remove this.

File: src/app/components/challenge-create/challenge-create.component.html [hunk 2]
```diff
@@ -1,27 +1,56 @@
...
+          </div>
+        </div>
+      </section>
 
     </div>
+    <app-footer [isDash]="true" *ngIf="authService.isAuth"></app-footer>
   </div>
 </div>
-<app-footer></app-footer>
+<div class="clearfix"></div>
```
Comment (lunayach) [inline src/app/components/challenge-create/challenge-create.component.html]: Is this needed?

File: src/app/components/challenge-create/challenge-create.component.ts [hunk 3]
```diff
@@ -73,33 +83,51 @@ export class ChallengeCreateComponent implements OnInit {
...
+          this.globalService.stopLoader();
+          this.globalService.showToast('error', err.error.error);
+          this.isSyntaxErrorInYamlFile = true;
+          this.syntaxErrorInYamlFile = err.error.error;
+        },
+        () => {}
+        );
+    } else {
+      this.isFormError = true;
+      this.globalService.showToast('error', 'Please Upload File');
```
Comment (lunayach) [inline src/app/components/challenge-create/challenge-create.component.ts]: The position of this is not aligned correctly. Render it directly below the `Upload file` text field.

File: src/app/components/challenge-create/challenge-create.component.html [hunk 4]
```diff
@@ -1,27 +1,56 @@
...
+
+                <div class="file-path-wrapper">
+                  <input disabled class="file-path validate" name="file_path" type="text">
+                </div>
+
+                <div class="align-left reg-control">
+                  <button class="btn ev-btn-dark waves-effect waves-dark grad-btn grad-btn-dark fs-14" type="submit" value="Submit">Submit</button>
+                </div>
+
+                <div *ngIf="isFormError" class="wrn-msg text-highlight align-right"> Please Upload File </div>
```
Comment (lunayach) [inline src/app/components/challenge-create/challenge-create.component.html]: The position of this is not aligned correctly. Render it directly below the `Upload file` text field.

**Review by lunayach — CHANGES_REQUESTED** [review]
The functionality looks good. Minor changes needed.

File: src/app/components/challenge-create/challenge-create.component.html (hunk 4 above)
Comment (sanketbansal) [reply src/app/components/challenge-create/challenge-create.component.html]: Done

File: src/app/components/challenge-create/challenge-create.component.html (hunk 2 above)
Comment (sanketbansal) [reply src/app/components/challenge-create/challenge-create.component.html]: Removed

File: src/app/components/challenge-create/challenge-create.component.scss (hunk 1 above)
Comment (sanketbansal) [reply src/app/components/challenge-create/challenge-create.component.scss]: Done

**Review by lunayach — APPROVED** [review]
@RishabhJain2018, Looks good to be merged.

**Review by RishabhJain2018 — APPROVED** [review]

**Pull request merged**